include/maths/CDataFrameCategoryEncoder.h

```diff
@@ -113,17 +113,23 @@ class MATHS_EXPORT CDataFrameCategoryEncoder final {
         double mic() const;
 
         //! Persist by passing information to \p inserter.
-        virtual void acceptPersistInserter(core::CStatePersistInserter& inserter) const;
+        void acceptPersistInserter(core::CStatePersistInserter& inserter) const;
 
         //! Populate the object from serialized data.
-        virtual bool acceptRestoreTraverser(core::CStateRestoreTraverser& traverser);
+        bool acceptRestoreTraverser(core::CStateRestoreTraverser& traverser);
 
         //! return encoding type as string
         virtual std::string typeString() const = 0;
 
     protected:
         std::size_t m_InputColumnIndex;
         double m_Mic;
+
+    private:
+        virtual void
+        acceptPersistInserterSubroutine(core::CStatePersistInserter& inserter) const = 0;
```

Comment:
I'm not sure about `Subroutine` here, which implies it could be doing anything. I think it would be nice to capture the fact that this is persisting the state for derived types. How about something like `acceptPersistInserterForDerivedTypeState` or something similar. Same comment applies for restore.

Reply:
Renamed